source/MaterialXGenGlsl/Nodes/GeomPropValueNodeGlsl.cpp

```diff
@@ -20,22 +20,78 @@ void GeomPropValueNodeGlsl::createVariables(const ShaderNode& node, GenContext&,
     const ShaderInput* geomPropInput = node.getInput(GEOMPROP);
     if (!geomPropInput || !geomPropInput->getValue())
     {
-        throw ExceptionShaderGenError("No 'geomprop' parameter found on geompropvalue node '" + node.getName() + "', don't know what property to bind");
+        throw ExceptionShaderGenError("No 'geomprop' parameter found on geompropvalue node '" + node.getName() + "'. Don't know what property to bind");
+    }
+    const string geomProp = geomPropInput->getValue()->getValueString();
+    const ShaderOutput* output = node.getOutput();
+
+    ShaderStage& vs = shader.getStage(Stage::VERTEX);
+    ShaderStage& ps = shader.getStage(Stage::PIXEL);
+
+    addStageInput(HW::VERTEX_INPUTS, output->getType(), HW::T_IN_GEOMPROP + "_" + geomProp, vs);
+    addStageConnector(HW::VERTEX_DATA, output->getType(), HW::T_IN_GEOMPROP + "_" + geomProp, vs, ps);
```

Comment:
Systematic use of T_IN_GEOMPROP insures the parameter ends up being named `i_geomprop_*` when varying and `u_geomprop_*` when uniform.